src/jit-format/jit-format.cs

```diff
@@ -0,0 +1,682 @@
+                        if (verbose)
+                        {
+                            Console.WriteLine("Building compile_commands.json.");
+                        }
+                        ParseBuildLog(config.BuildLog, compileCommands, project, config.SourceDirectory, verbose);
+                    }
+
+                    if (verbose)
+                    {
+                        Console.WriteLine("Using compile_commands.json found at {0}", compileCommands);
+                        Console.WriteLine("Running clang-tidy.");
+                    }
+
+                    if (!RunClangTidy(filenames, compileCommands, config.Fix, verbose))
+                    {
+                        Console.WriteLine("Clang-Tidy needs to be rerun in fix mode for {0}/{1}.", project, config.SourceDirectory);
+                        returncode = -1;
+                    }
+                }
+            }
+
+            if (config.DoClangFormat)
+            {
+                if (!RunClangFormat(filenames, config.Fix, verbose))
+                {
+                    Console.WriteLine("Clang-Format needs to be rerun in fix mode");
+                    returncode = -1;
+                }
+            }
+
+            return returncode;
+        }
+
+        public static void ParseBuildLog(string buildLog, string compileCommands, string project, string dir, bool verbose)
+        {
+            if (!File.Exists(buildLog))
+            {
+                Console.Error.WriteLine("Build log does not exist: {0}", buildLog);
+            }
+            using (var outputStream = System.IO.File.Create(compileCommands))
+            {
+                using (StreamWriter compileCommandsWriter = new StreamWriter(outputStream))
+                {
+                    compileCommandsWriter.WriteLine("[");
+                }
+            }
+
+            bool first = true;
+
+            foreach (string line in File.ReadLines(buildLog))
+            {
+                string clExe = "";
+                List<string> iOptions = new List<string>();
+                List<string> dOptions = new List<string>();
+                List<string> uOptions = new List<string>();
+                List<string> files = new List<string>();
+
+                // For now, only create the compile_commands database for the dll build of coreclr.
+                if (line.Contains("CL.exe") && line.Contains("src\\" + dir + "\\" + project))
+                {
+                    string[] splitLine = line.Trim().Split(new string[]{ "CL.exe " }, StringSplitOptions.None);
+                    clExe = (splitLine[0] + "CL.exe").Replace("\\", "/");
+
+                    string[] options = splitLine[1].Split(' ');
+
+                    bool dOptionFound = false;
+
+                    foreach (string option in options)
+                    {
+                        if (option.StartsWith("/I"))
```

Comment:
>/ [](start = 47, length = 1)

cl.exe accepts options beginning with "-" as well as "/" so you should check for both even if it happens that we don't see it today.